zpay32/invoice.go

```diff
@@ -71,12 +72,31 @@ const (
 	// fieldType9 contains one or more bytes for signaling features
 	// supported or required by the receiver.
 	fieldType9 = 5
+
+	// maxRouteHints is the maximum number of route hints allowed in a single
+	// invoice.
+	maxRouteHints = 20
```

Comment:
20 seems like a reasonable choice. Any reason in particular? :)

Reply:
Was the value previously used.  Currently the codebase only generates one hop per route hint so any invoice that lnd generates should fit into a qr code with 20.  However, if we stick with 20 here, it's possible that invoices could be larger-than-fits-in-qr-code while decoding.  Could lower `maxHopHints` since 12 hops for a route hint might be overkill or could lower `maxRouteHints` if 20 route hints is overkill.  Not sure

Reply:
> Was the value previously used. 
When LND creates invoices you mean? Could be useful share this constant with that code.

Reply:
I think as long as we keep a max total invoice length, maybe the other restrictions aren't even needed?

Reply:
Yup agree

> > Was the value previously used.
> 
> When LND creates invoices you mean? Could be useful share this constant with that code.

Yeah both when creating invoices + decoding invoices, but can just check the length now

Reply:
Done, now check that invoice is not greater than 7092 bytes.  Removed `maxRouteHints` but kept `maxHopHints` as it may be useful for callers of `Encode` to know why a specific invoice can't be serialized.

Reply:
Ah I see, every tagged field has a limit of 10 bits, but we don't seem to perform this check for any of them except `r` as of this PR.

Reply:
I think we check that all data will fit within 10 bits when encoding the invoice, so that should be enough for those cases?

I'm kinda wondering if we can just skip the `maxHopHints` check as well since we'll get the error when trying to fit within 10 bits 😛 What do you think @Crypt-iQ?

Reply:
Yeah I'm cool w that